optimum/neuron/modeling_diffusion.py

```diff
@@ -0,0 +1,501 @@
+        auto_cast_type = None if auto_cast is None else auto_cast_type
+        compiler_kwargs = {
+            "auto_cast": auto_cast,
+            "auto_cast_type": auto_cast_type,
+            "disable_fast_relayout": disable_fast_relayout,
+            "disable_fallback": disable_fallback,
+        }
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            compiler_kwargs=compiler_kwargs,
+            task=task,
+            dynamic_batch_size=dynamic_batch_size,
+            cache_dir=cache_dir,
+            trust_remote_code=trust_remote_code,
+            subfolder=subfolder,
+            revision=revision,
+            force_download=force_download,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            do_validation=False,
+            **input_shapes,
+        )
+
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            model_save_dir=save_dir,
+        )
+
+    @classmethod
+    def _load_config(cls, config_name_or_path: Union[str, os.PathLike], **kwargs):
+        return cls.load_config(config_name_or_path, **kwargs)
+
+    def _save_config(self, save_directory):
+        self.save_config(save_directory)
+
+
+class _NeuronDiffusionModelPart(NeuronBaseModel):
+    """
+    For multi-file Neuron models, represents a part of the model.
+    """
+
+    def __init__(
+        self,
+        model: torch.jit._script.ScriptModule,
+        parent_model: NeuronBaseModel,
+        config: Optional[Union[DiffusersPretrainedConfig, PretrainedConfig]] = None,
+        neuron_config: Optional["NeuronConfig"] = None,
+        model_type: str = "unet",
+        device: Optional[int] = None,
+    ):
+        self.model = model
+        self.parent_model = parent_model
+        self.config = config
+        self.neuron_config = neuron_config
+        self.model_type = model_type
+        self.device = device
+
+    @abstractmethod
+    def forward(self, *args, **kwargs):
+        pass
+
+    def __call__(self, *args, **kwargs):
+        return self.forward(*args, **kwargs)
+
+
+class NeuronModelTextEncoder(_NeuronDiffusionModelPart):
+    def __init__(
+        self,
+        model: torch.jit._script.ScriptModule,
+        parent_model: NeuronBaseModel,
+        config: Optional[DiffusersPretrainedConfig] = None,
+        neuron_config: Optional[Dict[str, str]] = None,
+    ):
+        super().__init__(model, parent_model, config, neuron_config, DIFFUSION_MODEL_TEXT_ENCODER_NAME)
+
+    def forward(self, input_ids: torch.Tensor):
+
+        inputs = (input_ids,)
+        outputs = self.model(*inputs)
+        return outputs
+
+
+class NeuronModelUnet(_NeuronDiffusionModelPart):
+    def __init__(
+        self,
+        model: torch.jit._script.ScriptModule,
+        parent_model: NeuronBaseModel,
```

Comment:
Do you really need the back-reference to the parent_model ? I did not see where it is actually used ...

Reply:
Oh, it's true because I inherited `NeuronBaseModel` which made it not necessary to have it as an attribute... (I will remove the inheritance and keep the attribute so that I could use some methods of `NeuronBaseModel`  when it makes sense.